### PR TITLE
Fix loading of TIF 16bits grayscale files

### LINKED ISSUE #1105: Error reading TIF 16bit Grayscale files
## Issue description
Reported in https://github.com/axondeepseg/axondeepseg/issues/617.
The current code for reading TIF files in the ivadomed loader [here](https://github.com/ivadomed/ivadomed/blob/1a3d193922938dc35e9eb77d6af7fd8e6d1bacfa/ivadomed/loader/segmentation_pair.py#L313) cannot read TIF 16bit Grayscale files.

### Current behavior
TIF 16bit Grayscale cannot be read with the `as_gray=True` arguments in `imageio` with the `tif-pill` plugin.
Every other combination of PNG/TIF, 8/16 bits, Grayscale/RGB/RGBA is working without issue.
ADS `imread` function has a fix for this [here](https://github.com/axondeepseg/axondeepseg/blob/ca4baad96bed0d9c84b2dc63a8c4b464dc4d5a2c/AxonDeepSeg/ads_utils.py#L259), which we should apply as well.

### PR DESCRIPTION
**Pull request by mariehbourget**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [x] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description
This PR fixes issue #1105 where TIF 16bits grayscale gave an error while loading:
- It includes a new test to make sure every combination of PNG/TIF, 8/16 bits, Grayscale/RGB/RGBA files are loaded properly.
- It requires a new set of testing images being integrated in data-testing in PR https://github.com/ivadomed/data-testing/pull/18.
- PR https://github.com/ivadomed/data-testing/pull/18 needs to be approved, merged and a new release of data-testing done to update the link for download [here](https://github.com/ivadomed/ivadomed/blob/6e1e87b4251a1e8985901194a821fb638d9242cb/ivadomed/scripts/download_data.py#L26) before merging the current PR.
- I updated some keywords in the code of the test_loader.py file as well.

## Linked issues
Fixes #1105

### CODE REVIEW

File: testing/unit_tests/test_loader.py [hunk 1]
```diff
@@ -37,8 +39,8 @@ def test_bids_df_microscopy_png(download_data_testing_test_files, loader_paramet
     bids_df = BidsDataframe(loader_parameters, __tmp_dir__, derivatives=True)
     df_test = bids_df.df.drop(columns=['path'])
     df_test = df_test.sort_values(by=['filename']).reset_index(drop=True)
-    csv_ref = Path(loader_parameters["path_data"][0], "df_ref.csv")
-    csv_test = Path(loader_parameters["path_data"][0], "df_test.csv")
+    csv_ref = Path(loader_parameters[LoaderParamsKW.PATH_DATA][0], "df_ref.csv")
+    csv_test = Path(loader_parameters[LoaderParamsKW.PATH_DATA][0], "df_test.csv")
```
Comment (dyt811) [inline testing/unit_tests/test_loader.py:43]: 👍

**Review by dyt811 — COMMENTED** [review]
Thanks for the concise PR. I will review with the PR18 data and approve it when it passes.

**Comment by akraskovSMDB** [comment]
Hi, I think I still have a problem with 16 bit tifs. Is there something wrong with my tif?

I run exactly the same training/testing using the same set of 16 and 8bit images and predictions of the model are wrong for 16 bit images but reasonable for 8 bit images. 

I have fresh install of ivadomed
pip show ivadomed
Name: ivadomed
Version: 2.9.8
Here is the [link](https://newcastle-my.sharepoint.com/:f:/g/personal/nak179_newcastle_ac_uk/EtLD8hsDgsxNh3alZDbocTUBsAw2DhuTTNNjtuS_kV6BtA?e=nNGv2f) to the images (I can't upload 16 bit tif, it is more than 25 Mb limit)

Happy to share whole BIDS folder and imadomed output folder with the config file

**Comment by akraskovSMDB** [comment]
Just checked and I have the same problem with 16 bit PNG images.

**Pull request merged**